### PR TITLE
Ipc fault artifacts

### PR DESCRIPTION
**Pull request by NamiLiy**
Artifacts needed to generate the 13 reproducable faults in MINIX 3 and Theseus

### CODE REVIEW

File: osdi20ae/ipc_fault_comparison/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# IPC fault comparison between Theseus and MINIX3
+
+This folder contains necessary components to evaluate fault recovery in IPC channels of Theseus and MINIX3
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: nit: missing period

File: osdi20ae/ipc_fault_comparison/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+
+## Evaluation Process on Theseus
+
+Run [script.sh](./script.sh) to generate the results, as described above.
+
+If one wanted to generate the same results from source without prebuilt images (reproduce the results), simply run the `./script_modify` and afterwards build theseus using the flags given above. Upon booting Theseus, run the following commands in Theseus's shell.
+
+   `ipc_faults -FAULT_ID`
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: put angle brackets around FAULT_ID, e.g., `ipc_faults -<FAULT_ID>`

File: osdi20ae/ipc_fault_comparison/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Eg
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: should be `e.g.,`

File: osdi20ae/ipc_fault_comparison/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+   `ipc_faults -r 3` to generate fault listed as r3 in the table
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: why is there an `-r` arg in this example but not above? 

Should the above example be `ipc_faults -r <FAULT_ID>` ?

File: osdi20ae/ipc_fault_comparison/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+## Evaluation Process on MINIX 3
+
+Modified source code for MINIX 3 is available at  [minix_osdi_ae](https://github.com/theseus-os/minix_osdi_ae). The repository contain a separate branch for each fault based on fault id. To reproduce each fault checkout the branch and build MINIX 3 using `bash ./releasetools/x86_hdimage.sh` .
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: More clarity is needed. Say something like this:

The repository contains a separate branch for each fault, where each branch has the same name as that fault id, e.g., `r1`.

File: osdi20ae/ipc_fault_comparison/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+Eg
+
+   `ipc_faults -r 3` to generate fault listed as r3 in the table
+
+## Evaluation Process on MINIX 3
+
+Modified source code for MINIX 3 is available at  [minix_osdi_ae](https://github.com/theseus-os/minix_osdi_ae). The repository contain a separate branch for each fault based on fault id. To reproduce each fault checkout the branch and build MINIX 3 using `bash ./releasetools/x86_hdimage.sh` . 
+
+Run MINIX 3 using the following command.
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: It's unclear how or where I should run this command. I assume you first clone the minix repo (which you should add instructions for), then `cd` into that repo, then checkout that branch (add instructions for that too), and then run that giant command in the base directory of that minix repo. 
Make sure to be very explicit, and leave no room for doubt.

File: osdi20ae/ipc_fault_comparison/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+`cd ../obj.i386/destdir.i386/boot/minix/.temp && qemu-system-i386 --enable-kvm -m 256M -kernel kernel -append "rootdevname=c0d0p0" -initrd "mod01_ds,mod02_rs,mod03_pm,mod04_sched,mod05_vfs,mod06_memory,mod07_tty,mod08_mib,mod09_vm,mod10_pfs,mod11_mfs,mod12_init" -hda ../../../../../src/minix_x86.img`
+
+After booting up login to Minix 3 using `root` as username. The source is modified to inject each fault in the table listed above to the IPC channel between input process and tty process, when letter `x` is entered to the shell.
```
Comment (kevinaboos) [inline osdi20ae/ipc_fault_comparison/README.md]: A bit hard to read, try this: "After Minix 3 boots up, login using `root` as username."

Also, i don't understand the part about "letter x being entered" ... if the user needs to type X or press X then enter, then you should explicitly state that here.

**Review by kevinaboos — CHANGES_REQUESTED** [review]
suggested changes for the README

File: osdi20ae/ipc_fault_comparison/README.md (hunk 6 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Got it. Didn't pay much attention to this side. Let me correct this

File: osdi20ae/ipc_fault_comparison/README.md (hunk 1 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Corrected

File: osdi20ae/ipc_fault_comparison/README.md (hunk 2 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Corrected

File: osdi20ae/ipc_fault_comparison/README.md (hunk 3 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Thanks. Corrected

File: osdi20ae/ipc_fault_comparison/README.md (hunk 4 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Corrected to use --r

File: osdi20ae/ipc_fault_comparison/README.md (hunk 5 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Corrected

File: osdi20ae/ipc_fault_comparison/README.md (hunk 6 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Corrected

File: osdi20ae/ipc_fault_comparison/README.md (hunk 7 above)
Comment (NamiLiy) [reply osdi20ae/ipc_fault_comparison/README.md]: Added a step by step explanation

**Review by kevinaboos — APPROVED** [review]

**Pull request merged**